src/background/tab-manager.ts

```diff
@@ -190,6 +214,27 @@ export default class TabManager {
         });
     }
 
+    async invalidateTimestamp() {
```

Comment:
It looks like the purpose of this function is to send a message, but not just update a timestamp. Could you put it back to see what have changed?

Reply:
I renamed it back to `sendMessage()` and moved it in old location.